### PR TITLE
[DO-NOT-MERGE] Checking to see if pyright verifies third-party test cases against false negatives correctly

### PR DESCRIPTION
**Pull request by AlexWaygood**
If our test-case setup is working correctly, pyright should error out with a complaint about an unused `type: ignore` comment.

### CODE REVIEW

**Comment by AlexWaygood** [comment]
Anybody have any ideas how to get pyright to emit an error on the unused `type: ignore` here? None of my ideas seem to work (see each commit in isolation for what I've tried).

**Comment by Avasam** [comment]
Weird, locally on main:
`python .\tests\pyright_test.py .\stubs\invoke\@tests -p .\pyrightconfig.testcases.json` works as expected, but 
`python .\tests\pyright_test.py -p .\pyrightconfig.testcases.json`

**Comment by AlexWaygood** [comment]
Maybe it's something specific to do with how pyright-action invokes pyright

**Comment by Avasam** [comment]
> Maybe it's something specific to do with how pyright-action invokes pyright

It's the same if I call pyright directly: `pyright .\stubs\invoke\@tests -p .\pyrightconfig.testcases.json`

**Comment by Avasam** [comment]
Ok, the * path is just broken,
Use 
```
    "include": [
        "stubs/invoke/@tests/test_cases",
    ],
```
and it works

**Comment by AlexWaygood** [comment]
Amazing, thanks for the investigation! I'll be AFK for a few hours, but I'll see about filing a bug report to pyright later if you don't beat me to it ;)

**Comment by Avasam** [comment]
Nvm the bisect part, I celebrated too early and didn't realize the errors I was getting were from a different file. :/

But what I said about using the direct path working is still true. So if it's broken, it's been broken since *forever*. I've open an issue with a minimal reproduction: #4553

**Comment by Avasam** [comment]
A fix has been released! https://github.com/microsoft/pyright/releases/tag/1.1.292
Not sure if you'd like to use this PR or a new one to start clean. But adding `test_cases` subfolders to the base and stricter configs' exclude is a good call.

**Comment by AlexWaygood** [comment]
> A fix has been released! https://github.com/microsoft/pyright/releases/tag/1.1.292 Not sure if you'd like to use this PR or a new one to start clean. But adding `test_cases` subfolders to the base and stricter configs' exclude is a good call.

Brilliant. I opened #9646 for the pyright bump.

File: pyrightconfig.json [hunk 1]
```diff
@@ -5,6 +5,10 @@
         "stdlib",
         "stubs",
     ],
+    "exclude": [
+        // test cases use a custom config file
+        "stubs*/@tests/test_cases"
```
Comment (AlexWaygood) [inline pyrightconfig.json:10]:
```suggestion
        "stubs/*/@tests/test_cases"
```

**Comment by AlexWaygood** [comment]
Closing in favour of #9650

**Pull request closed without merging**